### ISSUE TITLE
Notepad++ hangs and then unable to click on notepad++ window anywhere.

### ISSUE THREAD

**Issue by robertoprog** [wakatime/notepadpp-wakatime#36]
**Description of the Issue**
Notepad++ hangs and then unable to click on notepad++ window anywhere.

**Steps to Reproduce the Issue**
Open Notepad++ Window.
Open any program & and after sometime Notepad++ hangs and show the process in "Suspended" state.
Only closing from the taskbar "Close Window" & reopening solves this problem and sometimes "Close window" also does not work then Only "End Task" from Task Manager do the trick.
If there are some unsaved file open, I do see the option to save file, But still Notepad++ remains unclickable.

This behavior only occurs after installing the wakatime plugin.

Reference: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/4295

This still happens to me.

**Expected Behavior**
Notepad++ should not hang and becomes unclickable.
Actual Behavior
Notepad++ hangs and then unable to click on notepad++ window anywhere.

**Comment by Dauch** [comment]
It happed to me too, please fix this.

**Comment by adilabd** [comment]
me too unable from 20 dyas to get wakatime working

**Comment by kaetemi** [comment]
Still occurring here as well.

**Comment by xerif** [comment]
Same for me. Ever since I installed WakaTime 4.1 plugin (via the N++'s Plugin Admin), Notepad++ keeps hanging.

When it hangs:
- N++ process has a regular `Running` label on the Task Manager's detailed view
- N++ doesn't overuse any resources:
    ![image](https://user-images.githubusercontent.com/1078209/87074289-ef6e1d80-c21e-11ea-993b-040ab49f15c5.png)
- N++'s title bar doesn't have `(Not Responding)` suffix or anything like that
- N++ doesn't respond to **any** mouse or keyboard input, I can't even move the window by dragging the title bar
- When another app's window is displayed over N++ window, it's impossible to bring N++ into focus by clicking on it, or clicking on the task bar, or clicking `Switch to`, `Bring to front`, or `Maximize` context-menu option in the Task Manager.
- After some time had passed since the hang, right-clicking N++ on the task bar shows an empty white rectangle instead of a context menu.
    - **Bonus**: The rectangle stays on the screen forever and breaks context menus of some other apps. For example, now right-clicking something in the Task Manager shows context-menu without text, and the text only appears after hovering over it:
        ![image](https://user-images.githubusercontent.com/1078209/87075260-5c35e780-c220-11ea-93b4-2dcc42ca13fb.png)
    - Left-clicking any other item on the task bar no longer switches to that app, instead just marks the icons with orange pulsating background:
        ![image](https://user-images.githubusercontent.com/1078209/87075547-cd759a80-c220-11ea-8d6f-34f6f83aa04d.png)
    - Everything goes back to normal as soon as the N++ process is killed via the Task Manager: white rectangle disappears, other app's context menus work normally again, and I'm able to use the task bar to switch to other apps again.
- There's nothing useful in the log file, and the last log message is from several hours ago, when N++ was started:
    ```
    [Wakatime Info 01:44:26 PM] Initializing WakaTime v4.1.0
    [Wakatime Debug 01:44:26 PM] Python found from embedded location: C:\Users\xxx\AppData\Roaming\WakaTime\python\pythonw
    [Wakatime Info 01:44:26 PM] Current wakatime-cli version is 13.0.7
    [Wakatime Info 01:44:26 PM] Checking for updates to wakatime-cli...
    [Wakatime Debug 01:44:26 PM] No proxy will be used. It's either not set or badly formatted.
    [Wakatime Info 01:44:26 PM] wakatime-cli is up to date.
    [Wakatime Info 01:44:26 PM] Finished initializing WakaTime v4.1.0
    ```

I'm not sure how to reproduce the issue with 100% guarantee. Sometimes the hang happens after Notepad++ with the WakaTime plugin has been opened for several hours, sometimes it's less than an hour.

However, every time the hang has happened I was trying to open a source code file in it either by drag-and-dropping it onto the N++ window, or by `Edit with Notepad++` explorer context-menu action. Every time it happened the file had opened, but N++ immediately froze.

Whatever is going on, it's weird as 💩. I use N++ on a daily basis, I have several other plugins installed, but the problems have started only after installing the WakaTime plugin.

If there's any more info I can provide to help debug the issue, let me know.

**Comment by Olby2000** [comment]
Hi. The same has been happening to me too. At random times I would minimise the NPP window and it would not open up again. Sometimes it takes a while to click on different options in the right-click menu (in the taskbar) like opening another document and all of a sudden it will work again and go into maximised state. Very hard to reproduce reliably. However it has been happening for about a year or so. This made me uninstall the plugin. I forgot I did it and tried to re-install it later and, surely, it started to lock up again. W10 64 bit. Other plugins I have installed are: Mime Tools, Npp Converter, NppExport and XML Tools. Uninstalling other plugins does not help. Seems to happen very consistently after minimising Npp for a long time or after the computer was hibernating/the user account has been locked. Thanks.

**Comment by rjsfigueiredo** [comment]
hello, don't know if it is the case but I manage to skip this problem, because it seems that i have an invisible modal window above the np++ so I did: (in Window 10)
focus np++ window
press alt+space and it pops the menu i set to close the window.
then np++ was available again.
hope it helps.

**Comment by gandarez** [comment]
A new version has been released, please update it and come back to me if the problem gets fixed. You can manually download it from [Releases](https://github.com/wakatime/notepadpp-wakatime/releases/tag/4.2.1) or wait until this [PR](https://github.com/notepad-plus-plus/nppPluginList/pull/237) gets approved and then you can download using Plugins Admin.

**Comment by kaetemi** [comment]
Still occurring.

**Comment by gandarez** [comment]
I can't reproduce here that behavior. Which version and platform are you using?

**Comment by kaetemi** [comment]
The build you linked, 64-bit.

**Comment by gandarez** [comment]
I meant which N++ version?

**Comment by kaetemi** [comment]
v7.8.1

**Comment by gandarez** [comment]
Try updating it to v7.9. Also the problem seems to no occur only with WakaTime plugin.

**Comment by oldi** [comment]
I found that when I load large ascii files (>20mb) and have my spellchecker plugin active, then it hangs as it tries to spell check all the entries in the file. I simply turned the automatic spellcheck off - there is this little "ABC" button. I also deinstalled Waka. But I do not think tha Waka is the cause, at least in my case.

**Comment by makuke1234** [comment]
Hi! I am using the latest Wakatime plugin (version 4.2.3) and Notepad++ version v8.4.6 and I am still having problems with the Wakatime plugin, after a while, Notepad++ just hangs like the OP described. Wakatime is the only plugin that has been installed in Notepad++, without it, Notepad++ runs fine. I am using Microsoft Windows 10 21H2 64-bit and also the Notepad++ installation is 64-bit. Note that I have been having this issue with Notepad++ since I started using Wakatime plugin about a year ago although it has been supposedly "fixed".

Steps to reproduce:

1. Open Notepad++ and start to edit any file.
2. Minimize the Notepad++ window, wait for some time doing other things in other applications.
3. When trying to return to Notepad++, the window has frozen.

*Edit:
The Wakatime 4.2.3 is the latest version directly available in the Notepad++ plugin store.

**Comment by makuke1234** [comment]
I tried to also install the latest Wakatime plugin version 4.3.0 and Notepad++ throws an error explaining that this is an ANSI plugin incompatible with Unicode Notepad++ installation:

![image](https://user-images.githubusercontent.com/62077026/197407176-ac7e65d8-9b9b-4f43-a487-ba7fbe015ebd.png)

*Edit:
AFAIK my Notepad++ installation is the latest release, installed it a few days ago.

**Comment by alanhamlett** [comment]
These issues related to the WakaTime plugin should be fixed in v5.1.1.
